azurerm/internal/services/datashare/tests/resource_arm_data_share_dataset_kusto_database_test.go

```diff
@@ -0,0 +1,130 @@
+package tests
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+)
+
+func TestAccAzureRMDataShareDataSetKustoDatabase_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_data_share_dataset_kusto_database", "test")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMDataShareDataSetDestroy("azurerm_data_share_dataset_kusto_database"),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMDataShareDataSetKustoDatabase_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDataShareDataSetExists(data.ResourceName),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "display_name"),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "kusto_cluster_location"),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMDataShareDataSetKustoDatabase_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_data_share_dataset_kusto_database", "test")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMDataShareDataSetDestroy("azurerm_data_share_dataset_kusto_database"),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMDataShareDataSetKustoDatabase_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDataShareDataSetExists(data.ResourceName),
+				),
+			},
+			data.RequiresImportErrorStep(testAccAzureRMDataShareDataSetKustoDatabase_requiresImport),
+		},
+	})
+}
+
+func testAccAzureRMDataShareDataSetKustoDatabase_template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctest-datashare-%[1]d"
+  location = "%[2]s"
+}
+
+resource "azurerm_data_share_account" "test" {
+  name                = "acctest-dsa-%[1]d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  identity {
+    type = "SystemAssigned"
+  }
+}
+
+resource "azurerm_data_share" "test" {
+  name       = "acctest_ds_%[1]d"
+  account_id = azurerm_data_share_account.test.id
+  kind       = "InPlace"
+}
+
+resource "azurerm_kusto_cluster" "test" {
+  name                = "acctestkc%[3]d"
```

Comment:
ara capitals allowed here?
```suggestion
  name                = "acctestKC%[3]d"
```

Reply:
Not yet. Uppercase is not allowed in `azurerm_kusto_cluter` `name`.